### PR TITLE
Frontend update

### PR DESCRIPTION
**Pull request by aykl**

### CODE REVIEW

**Comment by gijskant** [comment]
Please update to `<node.version>v14.16.1</node.version>` in [pom.xml:79](https://github.com/thehyve/podium/blob/update/frontend/pom.xml#L79) to fix the build.

File: podium-gateway/package.json [hunk 1]
```diff
@@ -92,8 +92,8 @@
     "ts-node": "3.3.0",
     "tslint": "5.5.0",
     "tslint-loader": "3.5.3",
-    "typescript": "2.6.2",
-    "webpack": "3.6.0",
+    "typescript": "4.1.5",
+    "webpack": "4.46.0",
```
Comment (gijskant) [inline podium-gateway/package.json:83]: Looks like a dependency on `webpack-cli` now needs to be added:
```suggestion
    "webpack": "4.46.0",
    "webpack-cli": "^4.6.0",
```

File: .travis.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ cache:
 
 env:
   global:
-    - NODE_VERSION=10.20.0
+    - NODE_VERSION=15
```
Comment (gijskant) [inline .travis.yml]:
```suggestion
    - NODE_VERSION=14.16.1
```

File: podium-gateway/src/main/webapp/app/request/main-detail/detail/request-detail.component.html [hunk 3]
```diff
@@ -12,54 +12,54 @@
...
+                <dt><span [translate]="'request.detail.title'"></span></dt>
                 <dd>
                     <strong><span class="test-title">{{requestDetails.title}}</span></strong>
                 </dd>
-                <dt><span jhiTranslate="request.detail.organisation">Organisation</span></dt>
+                <dt><span [translate]="'request.detail.organisation'"></span></dt>
                 <dd>
                     <span class="test-organisations" *ngFor="let organisation of request.organisations">
-                        {{ organisation.name | capitalize }}
+                        {{ organisation.name | titlecase }}
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/app/request/main-detail/detail/request-detail.component.html]: Title case is not the same as capitalise. In the case of organisation name, these are not needed anyway.
```suggestion
                        {{ organisation.name }}
```

File: podium-gateway/src/main/webapp/app/request/main-detail/detail/request-detail.component.html [hunk 4]
```diff
@@ -93,14 +93,14 @@ <h3 jhiTranslate="request.relatedRequests">Related requests</h3>
...
-                                        {{ requestType | capitalize }}
+                                        {{ requestType | titlecase }}
                                     </li>
                                 </ul>
 
                             </td>
                             <td>
                                 <span *ngFor="let organisation of relatedRequest.organisations">
-                                    {{ organisation.name | capitalize }}
+                                    {{ organisation.name | titlecase }}
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/app/request/main-detail/detail/request-detail.component.html]:
```suggestion
                                    {{ organisation.name }}
```

File: podium-gateway/src/main/webapp/app/request/main-detail/detail/request-detail.component.html [hunk 5]
```diff
@@ -93,14 +93,14 @@ <h3 jhiTranslate="request.relatedRequests">Related requests</h3>
                             <td>
                                 <ul>
                                     <li *ngFor="let requestType of relatedRequest.requestDetail.requestType">
-                                        {{ requestType | capitalize }}
+                                        {{ requestType | titlecase }}
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/app/request/main-detail/detail/request-detail.component.html]: The types are also already capitalised, no need to do it here.
```suggestion
                                        {{ requestType }}
```

File: podium-gateway/src/main/webapp/main.ts [hunk 6]
```diff
@@ -19,6 +19,6 @@ if (!DEBUG_INFO_ENABLED) {
 }
 
 platformBrowserDynamic()
-    .bootstrapModule(PodiumGatewayAppModule, { preserveWhitespaces: true })
+    .bootstrapModule(PodiumGatewayAppModule, { preserveWhitespaces: false })
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/main.ts]: Please revert, this breaks spacing of buttons, e.g.:
Podium button spacing
The issue with the request type icons in the request list can be solved by removing `white-space: pre` from request-overview.scss.

File: podium-gateway/src/main/webapp/app/admin/health/modal/health-modal.component.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+  constructor(public activeModal: NgbActiveModal) {}
+
+  readableValue(value: any): string {
+    if (this.health?.key === 'diskSpace') {
+      // Should display storage space in an human readable unit
+      const val = value / 1073741824;
+      if (val > 1) {
+        return val.toFixed(2) + ' GB';
+      } else {
+        return (value / 1048576).toFixed(2) + ' MB';
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/app/admin/health/modal/health-modal.component.ts]: Hm, interesting that not ISO standards are followed here. This is from jhipster, right?
Correct would be:
```suggestion
        return (value / 1000000).toFixed(2) + ' MB';
```

File: podium-gateway/src/main/webapp/app/admin/health/modal/health-modal.component.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+})
+export class HealthModalComponent {
+  health?: { key: HealthKey; value: HealthDetails };
+
+  constructor(public activeModal: NgbActiveModal) {}
+
+  readableValue(value: any): string {
+    if (this.health?.key === 'diskSpace') {
+      // Should display storage space in an human readable unit
+      const val = value / 1073741824;
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/app/admin/health/modal/health-modal.component.ts:18]:
```suggestion
      const val = value / 1073741824;
```
```suggestion
      const val = value / 1000000000;
```

File: podium-gateway/src/main/webapp/app/admin/health/modal/health-modal.component.ts (hunk 7 above)
Comment (aykl) [reply podium-gateway/src/main/webapp/app/admin/health/modal/health-modal.component.ts]: Right, these admin components are auto-generated and have little (if any) changes from my side.

File: podium-gateway/src/main/webapp/app/core/util/alert.service.ts [hunk 9]
```diff
@@ -91,7 +90,7 @@ export class AlertService {
     private factory(alertOptions: Alert): Alert {
         const alert: Alert = {
             type: alertOptions.type,
-            msg: this.sanitizer.sanitize(SecurityContext.HTML, alertOptions.msg),
```
Comment (gijskant) [inline podium-gateway/src/main/webapp/app/core/util/alert.service.ts]: Why? Is this safe?

**Comment by gijskant** [comment]
- Building with maven will not correctly include the static front-end assets
- When entering Podium, the user is not redirected to the correct default page (my requests for researchers, my reviews for reviewers, etc)
- The alerts of the alert service don't show

There are quite some issues. And with so many changes it is difficult to locate what change caused the issue. Let's look at the issues next week.

File: podium-gateway/package.json [hunk 10]
```diff
@@ -7,115 +7,104 @@
...
+    "lint:fix": "npm run lint -- --fix",
     "start": "npm run webpack:dev",
-    "webpack:build": "webpack --config webpack/webpack.vendor.js && webpack --config webpack/webpack.dev.js",
-    "webpack:build:dev": "webpack --config webpack/webpack.dev.js",
-    "webpack:dev": "webpack-dev-server --config webpack/webpack.dev.js  --progress --inline --hot --profile --port=9060",
-    "webpack:prod": "npm test && webpack -p --config webpack/webpack.vendor.js && webpack -p --config webpack/webpack.prod.js",
-    "test": "npm run lint && karma start src/test/javascript/karma.conf.js",
-    "test:watch": "karma start --watch",
-    "postinstall": "webdriver-manager update && npm run webpack:build"
+    "webpack:build": "npm run clean-www && npm run webapp:build:dev",
```
Comment (gijskant) [inline podium-gateway/package.json]:
```suggestion
    "webpack:build": "npm run clean-www && npm run webpack:build:dev",
```

**Review by gijskant — APPROVED** [review]
Great work on the tests. Once rebased on `dev`, it is okay to merge.

**Pull request merged**